# Rule to delete unnecessary console.log (no-console-custom)

Please describe the origin of the rule here.


## Rule Details

This rule aims to...

Examples of **incorrect** code for this rule:

```js

// fill me in

```

Examples of **correct** code for this rule:

```js

// fill me in

```

### Options

If there are any options, describe them here. Otherwise, delete this section.

## When Not To Use It

Give a short description of when it would be appropriate to turn off this rule.

## Further Reading

If there are other links that describe the issue this rule addresses, please include them here in a bulleted list.

## Casos que queremos remover console.log

Se o pai dele for um bloco e o console não é o único elemento do bloco

https://astexplorer.net/#/gist/ab12d412ad08ea68f8dde21558721264/0232a3db7788347e7237ee20504dc93d269681b6

## triple equals

https://astexplorer.net/#/gist/b23c0f35c0800ab902a241093b7892c4/ba769b6b587a79d0d6f35fa5c6d63d02a0bbfb36

## simplificando expressões binárias

https://astexplorer.net/#/gist/60e8dc9cb4d46122c43b92f77f541896/30dfe4473bd9e42694b19d263f55bef5bd7ea650

## simplificando ifs que podem ser apenas um retorno

https://astexplorer.net/#/gist/fb47ca59f2ea99ab37d30c98f8e569d5/7326063a47cfd46d5cd9fadf0d34b5b7a7ea8a47

## simplificando metodos que retornam variavel local

https://astexplorer.net/#/gist/ebe267e0fab30c359654d13bfcc22c52/7ee314be7dd2656cd1a96b6fde4be12edf5d7abe

## simplificando if encadeados

https://astexplorer.net/#/gist/e3680e634a4777b5070048dad14d46b6/39efe22e30f035b51af2c90260d3e024932439a2